src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,82 @@
+package duke;
+import duke.commands.AddCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.OnDateCommand;
+import duke.commands.UnknownCommand;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+public class Parser {
+
+    /**
+     * Parses user input and responds with corresponding commands.
+     * 
+     * @param userInput input by user
+     * @return corresponding command of the input
+     * @throws DukeException
+     */
+    public static Command parse(String userInput) throws DukeException {
+        int firstBlankIndex = userInput.indexOf(" ");
+        String commandWord = firstBlankIndex == -1 ? userInput : userInput.substring(0, firstBlankIndex);
```

Comment:
This format is completely valid and compact. I think it would be easier for others to understand your code
if you expanded this statement, to something like:

```

String commandWord;
if (firstBlankIndex == -1) {
    commandWord = userInput;
} else {
    commandWord = userInput.substring(0, firstBlankIndex);
}
```

Reply:
I think this format would also have a similar effect.
```
String commandWord = firstBlankIndex == -1
    ? userInput
    : userInput.substring(0, firstBlankIndex);
```